### PR TITLE
[BIA-440] - Add DistrictId to UserAuthorization view

### PR DESCRIPTION
**Pull request by arturohernandez10**
For our use case we will have multiple districts using the middle tier authorization. But we don't have a simple way of knowing which is the district id of the user once we query the UserAuthorization view. Addind this column will give us this information that we need.

### CODE REVIEW

File: src/EdFi.AnalyticsMiddleTier.Tests/EdFi.AnalyticsMiddleTier.Tests.csproj [hunk 1]
```diff
@@ -1,5 +1,8 @@
 ﻿<Project Sdk="Microsoft.NET.Sdk">
-
+  <PropertyGroup>
+    <LangVersion>8.0</LangVersion>
+  </PropertyGroup>
+
```
Comment (stephenfuqua) [inline src/EdFi.AnalyticsMiddleTier.Tests/EdFi.AnalyticsMiddleTier.Tests.csproj:5]: Team should carefully evaluate before accepting a change in language. At the moment I'm not expressing an opinion, as I haven't given it any thought yet. Just want to dream the team's attention to this.

File: src/EdFi.AnalyticsMiddleTier.Tests/TestHarness.cs [hunk 2]
```diff
@@ -75,7 +75,7 @@ private TestHarness()
             // private so that this class cannot be instantiated elsewhere
         }
 
-        private string _connectionString => $"server=localhost;database={_databaseName};integrated security=sspi";
+        private string _connectionString => $"server=DESKTOP-A3LB20V,1434;database={_databaseName};user=sa;password=4$fdjewkQ";
```
Comment (stephenfuqua) [inline src/EdFi.AnalyticsMiddleTier.Tests/TestHarness.cs:78]: @arturohernandez10 please revert the change above.

File: src/EdFi.AnalyticsMiddleTier.Tests/TestHarness.cs [hunk 3]
```diff
@@ -229,7 +229,7 @@ public void ExecuteQuery(string sqlCommand)
 
         public void PrepareDatabase()
         {
-            using (var connection = new SqlConnection("server=localhost;database=master;integrated security=sspi"))
+            using (var connection = new SqlConnection("server=DESKTOP-A3LB20V,1434;database=master;user=sa;password=4$fdjewkQ"))
```
Comment (stephenfuqua) [inline src/EdFi.AnalyticsMiddleTier.Tests/TestHarness.cs:232]: Please revert

File: src/EdFi.AnalyticsMiddleTier.Tests/TestHarness.cs [hunk 4]
```diff
@@ -324,6 +324,7 @@ void CreateSnapshot()
 FROM pathname
 ";
                     var filePath = connection.ExecuteScalar<string>(defaultFilePathForSqlDbFiles);
+                    filePath = @"C:\Projects\IndianaUEdFi\Ed-Fi-Analytics-Middle-Tier\snapshots";
```
Comment (stephenfuqua) [inline src/EdFi.AnalyticsMiddleTier.Tests/TestHarness.cs:327]: Please revert.

**Comment by stephenfuqua** [comment]
General observation: looks great! Thank you for adding the unit tests. I noticed a few small but important things, but I'm leaving detailed review of the code to the Analytics team.

File: src/EdFi.AnalyticsMiddleTier.Tests/TestHarness.cs (hunk 4 above)
Comment (arturohernandez10) [reply src/EdFi.AnalyticsMiddleTier.Tests/TestHarness.cs:327]: @stephenfuqua Sorry about leaving extra code in here. I'll take it out. The language part, really was because I was not using the enterprise version of Visual Studio, and there was a part of the code that required 8.0. The project needed that to compile. I can try again without it so I can tell you what code was 8.0. I'll be making the changes shortly.

**Comment by arturohernandez10** [comment]
@stephenfuqua Sorry about leaving extra code in here. I'll take it out. The language part, really was because I was not using the enterprise version of Visual Studio, and there was a part of the code that required 8.0. The project needed that to compile. I can try again without it so I can tell you what code was 8.0. I'll be making the changes shortly.

**Comment by stephenfuqua** [comment]
> @stephenfuqua Sorry about leaving extra code in here. I'll take it out. The language part, really was because I was not using the enterprise version of Visual Studio, and there was a part of the code that required 8.0. The project needed that to compile. I can try again without it so I can tell you what code was 8.0. I'll be making the changes shortly.

Extra code: it happens, not a big deal 😃 . As for the language: I just double-checked and I see what you mean. Glad you were able to resolve with this update.

File: src/EdFi.AnalyticsMiddleTier.Tests/EdFi.AnalyticsMiddleTier.Tests.csproj (hunk 1 above)
Comment (stephenfuqua) [reply src/EdFi.AnalyticsMiddleTier.Tests/EdFi.AnalyticsMiddleTier.Tests.csproj:5]: On further review this looks appropriate. We were implicitly using language version 8.0 with VS2019. This just makes it more apparent and usable in other tools.

**Comment by arturohernandez10** [comment]
It's taking a few minutes to get the changes from our fork. I had improved on the tests too, you should see those updates as well.

File: src/EdFi.AnalyticsMiddleTier.Tests/TestHarness.cs [hunk 5]
```diff
@@ -229,7 +229,7 @@ public void ExecuteQuery(string sqlCommand)
 
         public void PrepareDatabase()
         {
-            using (var connection = new SqlConnection("server=localhost;database=master;integrated security=sspi"))
+            using (var connection = new SqlConnection("server=localhost;database={_databaseName};integrated security=sspi"))
```
Comment (jleiva-gap) [inline src/EdFi.AnalyticsMiddleTier.Tests/TestHarness.cs:232]: Please revert this change.  The PrepareDatabase connection string should use database = master.

File: src/EdFi.AnalyticsMiddleTier.DataStandard32/RLS/MSSQL/0007-View-UserAuthorization-Create.sql [hunk 6]
```diff
@@ -55,7 +55,11 @@ SELECT DISTINCT staffToScopeMap.StaffUniqueId AS UserKey
 		WHEN 'AuthorizationScope.District'
 			THEN 'ALL'
 		ELSE CAST(staffToScopeMap.EducationOrganizationId AS VARCHAR)
-		END AS SchoolPermission
+		END AS SchoolPermission,
+    CASE staffToScopeMap.UserScope
+        WHEN 'AuthorizationScope.District'
+            THEN staffToScopeMap.EducationOrganizationId
+        END AS DistrictId
```
Comment (jleiva-gap) [inline src/EdFi.AnalyticsMiddleTier.DataStandard32/RLS/MSSQL/0007-View-UserAuthorization-Create.sql:62]: For the DataStandard32, add these lines in the version for postgresql (RLS / PostgreSQL / 0007-View-rls_UserAuthorization-Create.rls)

Comment (DavidJGapCR) [reply src/EdFi.AnalyticsMiddleTier.DataStandard32/RLS/MSSQL/0007-View-UserAuthorization-Create.sql:62]: Yes, we need to add these changes to the Postgres version.

**Comment by DavidJGapCR** [comment]
Would it make sense to have a test scenario where the new field value returned is null?

**Comment by DavidJGapCR** [comment]
@arturohernandez10 , since this pull request has been open for a while, I just wanted to make sure you noticed that the Postgres script is still pending.

**Comment by arturohernandez10** [comment]
Ok, I guess I was hoping the comment was meant for someone else. Wishful thinking perhaps. I need to check with our project manager.

File: src/EdFi.AnalyticsMiddleTier.DataStandard32/RLS/MSSQL/0007-View-UserAuthorization-Create.sql (hunk 6 above)
Comment (david-clements-ed-fi) [reply src/EdFi.AnalyticsMiddleTier.DataStandard32/RLS/MSSQL/0007-View-UserAuthorization-Create.sql:62]: This is a good piece of feedback and initiated a quick internal discussion on how we should handle feature changes that affect multiple technology delivery options.  

Let's move this pull request into a feature branch and then the Ed-Fi team can make the modifications to include Postgres support.  We need the Postgres support added but don't want to add the burden to the contributing members who don't have that technology in their stack.

**Comment by stephenfuqua** [comment]
We have changed from having `development` and `master` branches to a single `main` branch. From `main` I've created a new feature branch and changed the PR to pull into that feature branch instead of directly into `main`. This gives the team more flexibility in accepting the PR. @david-clements-ed-fi  @arturohernandez10

File: src/EdFi.AnalyticsMiddleTier.DataStandard32/RLS/MSSQL/0007-View-UserAuthorization-Create.sql (hunk 6 above)
Comment (stephenfuqua) [reply src/EdFi.AnalyticsMiddleTier.DataStandard32/RLS/MSSQL/0007-View-UserAuthorization-Create.sql:62]: I'm going to accept the pull request into the feature branch. Thank you @arturohernandez10.
